File: app/src/main/java/com/example/phonequery/model/PhoneInfo.kt
package com.example.phonequery.model

/**
 * 统一的号码查询结果模型
 */
data class PhoneInfo(
    val number: String = "",
    val numberType: NumberType = NumberType.UNKNOWN,
    val province: String? = null,
    val city: String? = null,
    val carrier: String? = null,
    val areaCode: String? = null,
    val zipCode: String? = null,
    val spamType: String? = null,
    val spamCount: String? = null,
    val platformMarks: List<PlatformMark> = emptyList(),
    val codeNumberInfo: String? = null,
    val source: ResultSource = ResultSource.OFFLINE,
    val fromCache: Boolean = false,
    val errorMessage: String? = null
)

enum class NumberType(val displayName: String) {
    MOBILE("手机号码"),
    LANDLINE("固定电话"),
    TOLL_FREE("客服/热线"),
    UNKNOWN("未知类型")
}

enum class ResultSource(val displayName: String) {
    OFFLINE("本地离线"),
    ONLINE("在线接口"),
    CACHED("本地缓存")
}

data class PlatformMark(
    val platform: String,
    val mark: String
)

File: ios/RefusePhone/Core/PhoneInfo.swift
import Foundation

/// 号码分析结果（移植自 Android 版 PhoneInfo / SourceResult）
struct PhoneInfo {
    var number: String = ""
    var province: String?
    var city: String?
    var carrier: String?
    var areaCode: String?
    var zipCode: String?
    /// 骚扰提示（号段库/码号表命中）
    var spamHint: String?
    /// 码号资源信息（95/96/106/400/800 使用单位）
    var codeNumberInfo: String?
    /// 命中拦截规则
    var ruleHit: String?
    var isBlocked: Bool = false
    var isWhitelisted: Bool = false
    /// 最近一次在线查询错误信息（保留字段）
    var errorMessage: String?

    var summary: String {
        var parts: [String] = []
        if let p = province, !p.isEmpty { parts.append(p) }
        if let c = city, !c.isEmpty { parts.append(c) }
        if let ca = carrier, !ca.isEmpty { parts.append(ca) }
        return parts.joined(separator: " ")
    }
}
